### PR TITLE
docs: document metadata access for go-templated cluster generator

### PR DESCRIPTION
**Pull request by crenshaw-dev**

### CODE REVIEW

File: docs/operator-manual/applicationset/GoTemplate.md [hunk 1]
```diff
@@ -74,6 +74,13 @@ All your templates must replace parameters with GoTemplate Syntax:
 
 Example: `{{ some.value }}` becomes `{{ .some.value }}`
 
+### Cluster Generators
+
+By activating Go Templating, `{{ .metadata }}` becomes an object.
+
+- `{{ metadata.labels.my-label }}` becomes `{{ index .metadata.labels "my-label" }}`
+- `{{ metadata.annotations.my/annotation }}` becomes `{{ index .metadata.annotations "my/annotation" }}`
+
```
Comment (ishitasequeira) [inline docs/operator-manual/applicationset/GoTemplate.md:83]: @crenshaw-dev , was wondering if we need to provide a new example for this section. 
Or we could put this section below git generator and relate to the example present in the git generators

Comment (crenshaw-dev) [reply docs/operator-manual/applicationset/GoTemplate.md:83]: I think in this case I'd rather keep it as a simple "old format" vs. "new format" section. I'm not sure that there's anything to "demonstrate" with an example. The new syntax may be used anywhere the old syntax was used.

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Comment by ashutosh16** [comment]
LGTM

**Review by leoluz — APPROVED** [review]

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.5 for 2.5.2.

**Pull request merged**